### PR TITLE
[WIP] ELPP-2940 Add strict-transport-security header

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Refs https://github.com/elifesciences/journal/pull/637.

The `if` statement makes this complicated (has to be duplicated in every `location` directive). I'm tempted to say keep this in the application. It does mean that it wouldn't apply to assets, but that doesn't feel like a problem (as soon as you view an application-produced response the header will apply to the whole domain).

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
Thinking about it, the `if` could be done in Jinja, which means we could put it anywhere.

File: salt/journal/config/etc-nginx-traits.d-strict-transport-security.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+if ($https) {
+    add_header Strict-Transport-Security "max-age={{ 60 * 60 * 24 * 365 }}; includeSubDomains";
```
Comment (giorgiosironi) [inline salt/journal/config/etc-nginx-traits.d-strict-transport-security.conf]: `includeSubDomains` can be problematic - may break things like `api.elifesciences.org`, `some-crm.elifesciences.org`, `new-service-we-don-t-know-about.elifesciences.org`... that are not related to journal

**Comment by giorgiosironi** [comment]
I am tempted too because on the PHP side it will be more understandable. `if` in nginx configuration is a source of worries and problematic behavior, I have used it only in `iiif-formula` because I can't easily modify the Python code.

File: salt/journal/config/etc-nginx-traits.d-strict-transport-security.conf (hunk 1 above)
Comment (thewilkybarkid) [inline salt/journal/config/etc-nginx-traits.d-strict-transport-security.conf]: Do we have anything non-HTTPS still? Not having this blocks adding `preload`, which fixes a problem with the header (the initial redirect).

Comment (giorgiosironi) [inline salt/journal/config/etc-nginx-traits.d-strict-transport-security.conf]: http://demo--profiles.elifesciences.org/ (and every backend we expose temporarily for tests would need the SSL certificate) but unknown unknowns apply

Comment (thewilkybarkid) [inline salt/journal/config/etc-nginx-traits.d-strict-transport-security.conf]: It's not required for this anyhow, so let's move it to ELPP-2948.
